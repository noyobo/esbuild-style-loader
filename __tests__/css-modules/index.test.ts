import path from 'path';
import { runTest } from '../runTest';
import fse from 'fs-extra';

const OUTPUT_HTML = !!process.env.OUTPUT_HTML;

describe(path.basename(__filename), function () {
  it('style-loader', async function () {
    const output = path.resolve(__dirname, 'output');
    fse.removeSync(output);
    const result = await runTest([path.resolve(__dirname, './index.jsx')], output);
    const allFile = result.outputFiles.map((item) => item.path);
    allFile.forEach((file) => {
      fse.ensureFileSync(file);
      const content = result.outputFiles.find((item) => item.path === file)?.text;
      fse.writeFileSync(file, content);

      if (OUTPUT_HTML) {
        expect(1).toBe(1);
      } else {
        if (!file.endsWith('.map')) {
          expect(content).toMatchSnapshot();
        }
      }
    });
  });
});
