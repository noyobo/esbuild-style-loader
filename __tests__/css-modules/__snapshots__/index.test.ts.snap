// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.ts style-loader: __tests__/css-modules/output/index.css 1`] = `
"/* css-loader:./index.modules.css */
.index-modules__foo {
  color: red;
}
/*# sourceMappingURL=index.css.map */
"
`;

exports[`index.test.ts style-loader: __tests__/css-modules/output/index.js 1`] = `
"(() => {
  // style-loader:./index.modules.css
  var index_modules_default = { "foo": "index-modules__foo" };

  // __tests__/css-modules/index.js
  console.log(index_modules_default);
})();
//# sourceMappingURL=index.js.map
"
`;
